engine/kv_engine.hpp

```diff
@@ -188,11 +179,25 @@ class KVEngine : public Engine {
                  StringView elem) final;
   std::unique_ptr<ListIterator> ListMakeIterator(StringView key) final;
 
+  // Hash
+  Status HashLength(StringView key, size_t* len) final;
+  Status HashGet(StringView key, StringView field, std::string* value) final;
+  Status HashSet(StringView key, StringView field, StringView value) final;
+  Status HashDelete(StringView key, StringView field) final;
+  std::unique_ptr<HashIterator> HashMakeIterator(StringView key) final;
```

Comment:
I think external API should maintain same name style

Reply:
HashCreateIterator is also okay. But NewHashIterator is weird, as NewXXX means you have to delete it afterwards. Let's discuss it.
Make in HashMakeIterator comes from std::make_unique and std::make_shared.